STM32 target inherit: US_TICKER_MASK not defined

<!--

   ************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   Please ensure all sections of the template below are filled in and no changes 
   are made to the template format. Only bugs should be raised here as issues. 
   Questions or enhancements should instead be raised on our forums:
   https://forums.mbed.com/ .

   *************************************************************************************

-->

### Description of defect
When compiling a file below with target `-m NUCLEO_F412ZG` it compiles fine, but if I compile with `-m test` (custom target) the error below comes up.
<!--
    Add detailed description of what you are reporting.
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html
-->

```
Compile [ 99.0%]: hal_tick_overrides.c
[Error] hal_tick_overrides.c@53,56: 'US_TICKER_MASK' undeclared (first use in this function)
[ERROR] .\mbed-os\targets\TARGET_STM\hal_tick_overrides.c: In function 'HAL_GetTick':
.\mbed-os\targets\TARGET_STM\hal_tick_overrides.c:53:56: error: 'US_TICKER_MASK' undeclared (first use in this function)
   53 |     uint32_t elapsed_time = (((new_time - prev_time) & US_TICKER_MASK) + prev_tick_remainder);
      |                                                        ^~~~~~~~~~~~~~
.\mbed-os\targets\TARGET_STM\hal_tick_overrides.c:53:56: note: each undeclared identifier is reported only once for each function it appears in
```

#### Target(s) affected by this defect ?
Possibly all STM targets

#### Toolchain(s) (name and version) displaying this defect ?
GCC-ARM: 9.2.1 20191025

#### What version of Mbed-os are you using (tag or sha) ?
<!--
    For a released version please provide the release tag (this can be found as per the instructions below)

    mbed-os version can be found in /platform/mbed_version.h. The tag can be reconstructed as follows:
    mbed-os-MBED_MAJOR_VERSION.MBED_MINOR_VERSION.MBED_PATCH_VERSION
 
    Master branch is indicated by 'mbed-os-99.99.99
    
    For an issue found on Master please provide the sha being used.
-->
e345542b7005b9cc548e3c869f08813791ada442

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)
mbed-cli: 1.10.1

#### How is this defect reproduced ? 
`main.cpp`
```cpp
#include "mbed.h"

int main() {
    printf("test\n");
    return 0;
}
```
`custom_targets.json`
```json
{
    "test": {
        "inherits": ["NUCLEO_F412ZG"],
        "supported_form_factors": [],
        "overrides": {
            "clock_source": "USE_PLL_HSE_XTAL"
        }
    }
}
```

--- a/targets/TARGET_STM/device.h
+++ b/targets/TARGET_STM/device.h
@@ -37,6 +37,4 @@
 
 #include "objects.h"
 
-#include "us_ticker_defines.h"
-
 #endif

--- a/targets/TARGET_STM/hal_tick_overrides.c
+++ b/targets/TARGET_STM/hal_tick_overrides.c
@@ -17,7 +17,7 @@
 #if DEVICE_USTICKER
 
 #include "hal/us_ticker_api.h"
-#include "us_ticker_data.h"
+#include "us_ticker_defines.h"
 
 // This variable is set to 1 at the of mbed_sdk_init function.
 // The ticker_read_us function must not be called until the mbed_sdk_init is terminated.

